pkg/handlers/v1/nexposevulnnotifier.go

```diff
@@ -29,16 +30,20 @@ func (h *NexposeVulnNotificationHandler) Handle(ctx context.Context, in ScanInfo
 
 	assetChan, errChan := h.AssetFetcher.FetchAssets(ctx, in.SiteID)
 
-	// this loop currently logs when it receives an asset or error from their respective channels
-	// The future issues will take these assets and use them to call Nexpose again to get heir vulnerabilities
 	for {
 		select {
-		case _, ok := <-assetChan:
+		case asset, ok := <-assetChan:
 			if !ok {
 				assetChan = nil
 			} else {
-				// TODO this is where we'll publish the asset to the queue SECD-442
 				stater.Count("assetreceived.success", 1)
+				err := h.Producer.Produce(ctx, asset)
```

Comment:
note: assuming you are producing the SQS, the sendmessage latency is in the [tens to hundreds of ms](https://aws.amazon.com/sqs/faqs/). Let's say a typical sendmessage response time is 50ms. If you are expected assets to be in the thousands, this for loop will take minutes to complete.Since this lambda handler is intended to be async, I am just making note of this as something to be aware of, and not necessarily requesting changes to be made.

If you wanted to explore ways to improve this, I think there are a few things you could look into. The simplest, and least invasive would be wrapping the produce call in a goroutine.

```
go func(ctx context.Context, asset domain.Asset) {
    err := h.Producer.Produce(ctx, asset)
    if err != nil {
        stater.Count("producer.error", 1)
        logger.Error(logs.ProducerFailure{
            Reason: err.Error(),
	})
    }
}(ctx, asset)
```

This would be simple to do since the handler response doesn't depend on these errors. You can continue to log stat them as you are doing. With this approach you may also consider using a semaphore to limit the number of produce routines you spin up.

Reply:
I see. That's interesting. I didn't consider the time it would take for the message to be produced and having to wait for its response to be returned to the streaming appliance to then be returned here.  I'll add it, thanks!